mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -407,6 +449,99 @@ private[classification] class MultiClassSummarizer extends Serializable {
   }
 }
 
+@Experimental
+class LogisticRegressionTrainingSummary private[classification] (
+    predictions: DataFrame,
+    probabilityCol: String,
+    labelCol: String,
+    val objectiveHistory: Array[Double])
+  extends LogisticRegressionSummary(predictions, probabilityCol, labelCol) {
+
+  /** Number of training iterations until termination */
+  val totalIterations = objectiveHistory.length
+
+}
+
+@Experimental
+class LogisticRegressionSummary private[classification] (
+  @transient val predictions: DataFrame,
+  val probabilityCol: String,
+  val labelCol: String) extends Serializable {
+
+  /** Returns a BinaryClassificationMetrics object.
+  */
+  @transient val metrics = new BinaryClassificationMetrics(
+    predictions.select(probabilityCol, labelCol).map {
+      case Row(score: Vector, label: Double) => (score(1), label)
+    }
```

Comment:
Indeed. But how do we expose this to the user. Or do we use some sort of heuristic?